### PR TITLE
Alpine support

### PR DESCRIPTION
**Pull request by orangedeng**
Refer to issue: https://github.com/cnrancher/autok3s/issues/602
- Support install k3s on Alpine linux
- Refactor ssh package

### CODE REVIEW

File: pkg/hosts/dialer/ssh.go [hunk 1]
```diff
@@ -126,95 +130,147 @@ func (d *SSHDialer) Dial(t bool) (*ssh.Client, error) {
...
+	}
+	defer session.Close()
+
+	encodedCMD := d.wrapCommands(strings.Join(cmds, "\n"))
+	cmd := fmt.Sprintf("echo \"%s\" | base64 -d | %s sh -", encodedCMD, sudo)
+	d.logger.Debugf("executing cmd: %s", cmd)
+
+	output := bytes.NewBuffer([]byte{})
+	combinedOutput := singleWriter{
+		b: io.MultiWriter(d.logger.Out, output),
```
Comment (JacieChao) [inline pkg/hosts/dialer/ssh.go:196]: Where do we initialize the mutext lock for singleWriter?

Comment (orangedeng) [reply pkg/hosts/dialer/ssh.go:196]: Here is copy code from combine output, as we need to set combineOutput to both stdout and stderr.

**Review by JacieChao — APPROVED** [review]
LGTM

**Review by Jason-ZW — APPROVED** [review]
LGTM

**Pull request merged**